scipy/odr/__init__.py

```diff
@@ -22,11 +22,9 @@
    OdrError      -- Error exception.
    OdrStop       -- Stop exception.
 
-Prebuilt models:
-
-.. autosummary::
+   polynomial    -- Factory function for a general polynomial model.
 
-   polynomial
+Prebuilt models:
```

Comment:
This makes sense, the 4 objects below are `Model` instances, so it doesn't make sense to generate doc pages for them. Had to look at this for a second, but LGTM.

Reply:
Currently `polynomial` is a function of `odr`, but other 4 models are instances of Models. I think the other 4 models also should be functions which just return the model instances for consistency. If we did it, I believe other functions will have docs correctly. Should I do it in this PR?

Reply:
That would fix the issue, however it would be a backwards incompatible change, so better not in this PR. It's also unclear to me if the current implementations are useful/used. There doesn't seem to be an issue for this yet, could you open one?

Reply:
I did it in #12087